### PR TITLE
[MAINTENANCE] Rule-Based Profiler: Standardize on Include/Exclude Column Names List

### PR DESCRIPTION
**Pull request by alexsherstinsky**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
-
-
-

After submitting your PR, CI checks will run and @ge-cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py [hunk 1]
```diff
@@ -18,59 +18,70 @@ def __init__(
...
-        self._column_names = value
+    @property
+    def exclude_column_names(self) -> Optional[Union[str, Optional[List[str]]]]:
+        return self._exclude_column_names
+
+    @include_column_names.setter
+    def include_column_names(
+        self, value: Optional[Union[str, Optional[List[str]]]]
+    ) -> None:
+        self._include_column_names = value
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py:62]: Is there ever a case where we pass in a null value to the setter? If not, could we get rid of `Optional`?

File: great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py [hunk 2]
```diff
@@ -18,59 +18,70 @@ def __init__(
         batch_list: Optional[List[Batch]] = None,
         batch_request: Optional[Union[BatchRequest, RuntimeBatchRequest, dict]] = None,
         data_context: Optional["DataContext"] = None,  # noqa: F821
-        column_names: Optional[Union[str, Optional[List[str]]]] = None,
+        include_column_names: Optional[Union[str, Optional[List[str]]]] = None,
+        exclude_column_names: Optional[Union[str, Optional[List[str]]]] = None,
```
Comment (cdkini) [inline great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py:22]: Shouldn't this just be an `Optional[List]`? Why is this signature so complex?

**Review by cdkini — APPROVED** [review]
LGTM but please review my comment around the type of `include_column_names` and `exclude_column_names`.

If we can isolate the type down to a simple `Optional[List[str]]`, I'd really prefer that!

File: great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py:62]: can be accidentally in config

Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py:62]: @cdkini also could be in formal object instantiation

File: great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py:22]: @cdkini This `str` is to allow `$variable` syntax.  Thanks.

**Pull request merged**